### DISCUSSION TITLE
Using Voikko in CSC Puhti HPC

### DISCUSSION THREAD

**Discussion by mrgransky** [voikko/discussion#16] [Q&A]
Hello,

Apologies in advance if this question may seem irrelevant to the Voikko library!

I am trying to code some spell checking for Finnish and Swedish languages as part of my research.
I came along this library for Finnish language (as an alternative for Enchant).

In my own local ubuntu 22 laptop (with sudo privilege), I could simply install all dependencies by doing:

`$ sudo apt install enchant-2 libenchant-2-* libaspell* voikko-fi  aspell* -y`

Then in my conda virtual environment, I do the following:

```
$ pip install libvoikko # no conda alternative exists!
$ conda install -c conda-forge pyenchant enchant aspell
```

And both work successfully in my local machine:
![bild](https://github.com/voikko/discussion/assets/11946010/56bd3672-a3e9-4eb0-bd5c-740d7884e280)

Now doing the same procedure for the CSC Puhti supercomputer in Finland (to leverage higher computational resources), I do not have the sudo privilege to install dependencies in Red Hat Enterprise Linux release 8.6 (Ootpa) and I end up getting error for both Voikko and enchant:

![bild](https://github.com/voikko/discussion/assets/11946010/7d978322-a700-4e02-a86d-70a4067f8114)

Searching for available packages (Aspell, enchant, voikko and libenchant) in RedHat Puhti leads to:

![bild](https://github.com/voikko/discussion/assets/11946010/4bce03be-6800-4109-84fb-fab25f4a6229)

Has anybody else faced the same issue while using CSC supercomputer, e.g., Puhti in this case?

Or is there a better and easier solution to handle such issues when we do not have sudo privilege?

Cheers,

**Comment by hatapitk** [comment]
Hi! Sorry for very late answer. I guess compiling libvoikko from source would be the easiest way to solve this. It has no mandatory dependencies for Finnish (use --disable-hfst when compiling).
